.github/workflows/rabbit_consumer_schedule.yaml

```diff
@@ -0,0 +1,120 @@
+
+jobs:
+  test_and_lint:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.10"]
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: "pip"
+      - name: Install dependencies
+        run: |
+          sudo apt-get update --fix-missing
+          python -m pip install --upgrade pip
+          # Required for requests-kerberos
+          sudo DEBIAN_FRONTEND=noninteractive apt-get -yq install libkrb5-dev
+          pip install -r OpenStack-Rabbit-Consumer/requirements.txt
+          pip install -r OpenStack-Rabbit-Consumer/requirements-test.txt
+      
+      - name: Run tests
+        # Using Python3 to launch the module sets up the Python path for us
+        run: cd OpenStack-Rabbit-Consumer && python3 -m coverage run -m pytest .
+      
+      - name: Analyse with pylint
+        run: |
+          cd OpenStack-Rabbit-Consumer && pylint $(git ls-files '*.py')
+
+      - name: Prepare coverage
+        run: |
+          cd OpenStack-Rabbit-Consumer && python -m coverage xml
+
+  push_dev_image_harbor:
+    runs-on: ubuntu-latest
+    needs: test_and_lint
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Login to Harbor
+        uses: docker/login-action@v2
+        with:
+          registry: harbor.stfc.ac.uk
+          username: ${{ secrets.HARBOR_USERNAME }}
+          password: ${{ secrets.HARBOR_TOKEN }}
+
+      - name: Set commit SHA for later
+        id: commit_sha
+        run: echo "sha_short=$(git rev-parse --short HEAD)" >> $GITHUB_OUTPUT
+
+      - name: Build and push to staging project
+        uses: docker/build-push-action@v3
+        with:
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          push: true
+          context: "{{defaultContext}}:OpenStack-Rabbit-Consumer"
+          tags: "harbor.stfc.ac.uk/stfc-cloud-staging/openstack-rabbit-consumer:${{ steps.commit_sha.outputs.sha_short }}"
+
+      - name: Inform of tagged name
+        run: echo "Image published to harbor.stfc.ac.uk/stfc-cloud-staging/openstack-rabbit-consumer:${{ steps.commit_sha.outputs.sha_short }}"
+
+  push_release_image_harbor:
+    runs-on: ubuntu-latest
+    needs: test_and_lint
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Login to Harbor
+        uses: docker/login-action@v2
+        with:
+          registry: harbor.stfc.ac.uk
+          username: ${{ secrets.HARBOR_USERNAME }}
+          password: ${{ secrets.HARBOR_TOKEN }}
+
+        #increments ONLY the patch version using the version_incrememnt.sh script and commits the version.txt file to master
+      - name: Increment Version
```

Comment:
Is there a way we only increment version when our dependency versions change? 
Set here. https://github.com/stfc/SCD-OpenStack-Utils/blob/master/OpenStack-Rabbit-Consumer/requirements.txt 

To do this - we need to put versions against each of our dependencies which dependabot can update for us

Reply:
This makes sure that we're not updating the patch version with no changes being made

Reply:
Quite possibly, I didn't look into this but me and David did discuss ways to include conditionals depending on what files are being updated but decided against it as it could over complicate it for now. Something that we could definitely consider though.

Reply:
To follow-up I chatted to Anish about this, unfortunately the main target for this work is https://github.com/stfc/SCD-OpenStack-Utils/blob/1991246795d074d043eb76e58b78a9eee454ff09/OpenStack-Rabbit-Consumer/Dockerfile#L1 where we want to grab any security patches (or OS patches) for python 3.10

Because a label in Docker isn't static (unlike a tag in source control) 3.10 could refer to a base image with September patches one month, then October patches the next week but still be labelled `3.10`